services/eos-dev/contracts/pet/pet.cpp

```diff
@@ -261,60 +340,6 @@ void pet::_update(st_pets &pet) {
     }
 }
 
-// we need to sacrifice abi generation for recipient listener
-// keep alternating the comments between EOSIO_ABI (to generate ABI)
-// and EOSIO_ABI_EX to generate the listener action
-// https://eosio.stackexchange.com/q/421/54
-
-// EOSIO_ABI(pet, (createpet)(updatepet)(feedpet)(bedpet)(awakepet)(destroypet)(battlecreate)(battlejoin)(battleleave)(battlestart)(battleselpet)(battleattack)(battlefinish)(addelemttype)(changeelemtt)(addpettype)(changepettyp)(changecrtol)(changebatma)(changebatidt)(changebatami)(changebatama)(transfer))
-
-#define EOSIO_ABI_EX( TYPE, MEMBERS ) \
```

Comment:
loved your idea of removing the comments and splitting in two cpp files, one for regular compilation and other one for abi, nice job! 👍